chore: use goreman to start principal and agent

jparsai opened the pull request:
This PR is to make use of `goreman` to run principal and agent in same console, this would make debugging easy as logs of both components would be printed in same console, also it would help to understand the order of execution of agent and principal code blocks.

jannfis reviewed (commented):
Thanks, I do like this idea.

However, I think neither goreman nor make configuration should be part of the top-level for now. 

My suggestion would be to place the Procfile into (a new) `hack/demo-env/Procfile` and Makefile (if required at all) into `hack/demo-env/Makefile` instead.

jannfis reviewed (commented):
Also, I think that there really should be two agent targets in `Procfile`, `agent-managed` and `agent-autonomous`. Or some logic that lets you start one or the other.

jparsai commented:
Thanks for looking into this. I moved `Procfile` inside `hack/demo-env/` directory, but moving `Makefile` requires changes in start up scripts, because they have to be executed from root folder. I was not sure if I should modify start up scripts for this.
To set agent mode I introduced an env variable `AUTONOMOUS_MODE`, which can be used like this

`AUTONOMOUS_MODE=true make start-local
`

jannfis reviewed (approved):
LGTM!
